utils/src/com/cloud/utils/S3Utils.java

```diff
@@ -352,10 +352,15 @@ public static File getFile(final ClientOptions clientOptions, final String bucke
         ListObjectsRequest listObjectsRequest = new ListObjectsRequest().withBucketName(bucketName).withPrefix(directory + SEPARATOR);
 
         ObjectListing ol = client.listObjects(listObjectsRequest);
```

Comment:
we don't catch any of the amazon runtime exceptions. I think we should wrap them in a CloudstackRuntimeException so they can be caught be the servlet before propagating into the container. This is not in your PR @borisroman so I won't -1 on it but if you are on it, why not improve ;)